### PR TITLE
Fire info callbacks in all states and maintain state information

### PR DESCRIPTION
**Pull request by yeesian**
ref: https://github.com/JuliaOpt/JuMP.jl/pull/814

### CODE REVIEW

File: src/CplexSolverInterface.jl [hunk 1]
```diff
@@ -449,22 +449,32 @@ function mastercallback(env::Ptr{Void}, cbdata::Ptr{Void}, wherefrom::Cint, user
...
+
+    if model.infocb != nothing
+        cpxcb = CplexInfoCallbackData(cpxrawcb, state, wherefrom)
+        stat = model.infocb(cpxcb)
+        if stat == :Exit
+            terminate(model.inner)
+        end
+    end
+    if model.lazycb != nothing && state == :MIPSol
+        cpxcb = CplexLazyCallbackData(cpxrawcb, state, wherefrom, userinteraction_p)
```
Comment (joehuchette) [inline src/CplexSolverInterface.jl]: Could this overwrite the previous value for ``cpxcb`` if there is both an info callback, a lazy callback, and the ``state == :MIPSol``?

Comment (yeesian) [inline src/CplexSolverInterface.jl]: yup, it can. `cpxcb` here is a temporary alias within each of the `if` statements.

Comment (yeesian) [inline src/CplexSolverInterface.jl]: Should I prefix them with a `local`?

Comment (joehuchette) [inline src/CplexSolverInterface.jl]: Oh, nevermind, I missed line 461 where the callback function was actually called.

**Pull request merged**